servicetalk-client-api/src/main/java/io/servicetalk/client/api/DelegatingConnectionFactory.java

```diff
@@ -43,7 +43,18 @@ public class DelegatingConnectionFactory<ResolvedAddress, C extends ListenableAs
      * @param delegate {@link ConnectionFactory} to which all methods are delegated.
      */
     public DelegatingConnectionFactory(final ConnectionFactory<ResolvedAddress, C> delegate) {
-        this.delegate = requireNonNull(delegate);
+        super(delegate);
+        this.delegate = delegate;
+    }
+
+    /**
+     * Returns the {@link ConnectionFactory} delegate.
+     *
+     * @return Delegate {@link ConnectionFactory}.
+     */
+    @Override
+    protected final ConnectionFactory<ResolvedAddress, C> delegate() {
+        return delegate;
     }
```

Comment:
Not critical, but can we avoid this noise if we make `DelegatingListenableAsyncCloseable` generic on the type of delegate?

Reply:
I'm actually looking at `DelegatingSslConfig` and thinking about the opposite. Generics quickly become complex to use, especially when you need to use `Delegating*` class as a map value. With my recent experience in gRPC codegen, I would love to limit amount of generics we use everywhere.

Reply:
I will give it a try in a follow-up PR where we can see how it shapes. If we don't like it, we will close. Will submit shortly

Reply:
See #3245